pypsa/linopf.py

```diff
@@ -403,7 +403,9 @@ def bus_injection(c, attr, groupcol='bus', sign=1):
            .reindex(columns=n.buses.index, fill_value=''))
 
     if (lhs == '').any().any():
-        raise ValueError("Empty LHS in nodal balance constraint.")
+        logger.warning(f"Empty LHS in nodal balance constraint (unconnected buses?): {lhs[lhs=='']}")
+        lhs = lhs[~(lhs=='')]
```

Comment:
This selection is not applied to the right hand side, right? This might lead to complications if the demand of a bus without any other component is given.

Reply:
Yes - let's discard. I believe this is one function where a disconnected bus throws an exception, but it didn't resolve the issue. And, as you say, could cause other problems and would need to be checked.